icu4c/source/common/uresdata.cpp

```diff
@@ -963,7 +963,7 @@ res_findResource(const ResourceData *pResData, Resource r, char** path, const ch
       if(t2 == RES_BOGUS) { 
         /* if we fail to get the resource by key, maybe we got an index */
         indexR = uprv_strtol(pathP, &closeIndex, 10);
-        if(indexR >= 0 && *closeIndex == 0) {
+        if(indexR >= 0 && *closeIndex == 0 && (*pathP != '0' || (*pathP == '0' && closeIndex - pathP == 1))) {
```

Comment:
I don't think we need the *pathP == '0' test (if we process pass the || we already know *pathP == '0') , can't the added part just be
`&& (*pathP != '0' || closeIndex - pathP == 1)`